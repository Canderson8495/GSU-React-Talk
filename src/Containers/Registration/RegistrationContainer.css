.Registration-container {
    border-width: 1px;
    border-style: solid;
    padding: 10vh;
    display: inline-block;
    border-radius: 4px;
    border-color: #15f4ee;
}

.Registration-title {
    margin-bottom: 10vh;
    padding-inline: 10vh;
}

.Registration-button {
    color: white;
    background-color: transparent;
    border-radius: 4px;
    border-color: #15f4ee;
    border-style: solid;
    border-width: 1px;
    padding-inline: 32px;
    padding-block: 8px;
    font-size: 24px;
    margin-top: 24px;
}