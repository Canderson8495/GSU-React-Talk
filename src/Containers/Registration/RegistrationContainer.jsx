import './RegistrationContainer.css'
import RegistrationField from '../../Components/RegistrationField';
import { useState } from 'react';
import { validateBirthday, validateName, validatePassword, validateVerifyPassword, validateEmail } from '../../Common/RegistrationValidation';
function RegistrationContainer() {

    const [firstName, setFirstName] = useState({value: '', error: ''});
    const [lastName, setLastName] = useState({value: '', error: ''});
    const [email, setEmail] = useState({value: '', error: ''});
    const [password, setPassword] = useState({value: '', error: ''});
    const [verifyPassword, setVerifyPassword] = useState({value: '', error: ''});
    const [birthday, setBirthday] = useState({value: '', error: ''});

    const handleFirstNameChange = (event ) => {
        setFirstName({value: event.target.value, error: ''});
    }
    const handleLastNameChange = (event ) => {
        setLastName({value: event.target.value, error: ''});
    }
    const handleEmailChange = (event ) => {
        setEmail({value: event.target.value, error: ''});
    }
    const handlePassword = (event ) => {
        setPassword({value: event.target.value, error: ''});
    }
    const handleVerifyPasswordChange = (event ) => {
        setVerifyPassword({value: event.target.value, error: ''});
    }
    const handleBirthdayChange = (event ) => {
        setBirthday({value: event.target.value, error: ''});
    }

    const handleRegisterClick = () => {
        let failed = false;

        let error = validateName(firstName.value)
        if(error !== '' ){
            failed = true
            setFirstName({value: firstName.value, error:error})
        }

        error = validateName(lastName.value)
        if(error !== '' ){
            failed = true
            setLastName({value: lastName.value, error:error})
        }

        error = validateEmail(email.value)
        if(error !== '' ){
            failed = true
            setEmail({value: email.value, error:error})
        }

        error = validatePassword(password.value)
        if(error !== '' ){
            failed = true
            setPassword({value: password.value, error:error})
        }

        error = validateVerifyPassword(verifyPassword.value)
        if(error !==  ''){
            failed = true
            setVerifyPassword({value: verifyPassword.value, error: error})
        }

        error = validateBirthday(verifyPassword.value)
        if(error !==  ''){
            failed = true
            setBirthday({value: birthday.value, error: error})
        }

        if(!failed){
            console.log("Make a POST request here!")
        }

    }

    //HOMEWORK TO AUDIENCE: stage-3 branch | Why don't you try to use the useReducer hook to clean this up and consolidate this?
    return (
        <div className="Registration-container">
          <div className="Registration-title"> 
            <h1> Registration </h1>
          </div>
          <div className="Registration-fields">

            <RegistrationField
                name="fname"
                labelText="First Name"
                type="text"
                handler={handleFirstNameChange}
                data={firstName}
            />

            <RegistrationField
                name="lname"
                labelText="Last Name"
                type="text"
                handler={handleLastNameChange}
                data={lastName}
            />

            <RegistrationField
                name="email"
                labelText="Email"
                type="text"
                handler={handleEmailChange}
                data={email}
            />
            
            <RegistrationField
                name="password"
                labelText="Password"
                type="password"
                handler={handlePassword}
                data={password}
            />

            <RegistrationField
                name="vpassword"
                labelText="Verify Password"
                type="password"
                handler={handleVerifyPasswordChange}
                data={verifyPassword}
            />

            <RegistrationField
                name="bday"
                labelText="Birthday"
                type="text"
                handler={handleBirthdayChange}
                data={birthday}
            />

          </div>
          <button onClick={handleRegisterClick} className='Registration-button' type="button">
            <p> Register </p> 
            </button>
        </div>
    );
}

export default RegistrationContainer;