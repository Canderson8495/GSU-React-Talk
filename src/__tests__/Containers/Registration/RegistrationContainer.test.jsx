import { render, screen } from '@testing-library/react';
import RegistrationContainer from '../../../Containers/Registration/RegistrationContainer'
import userEvent from "@testing-library/user-event";


test('Snapshot test of Registration Container', () => {
  const tree = render(<RegistrationContainer />);
  expect(tree).toMatchSnapshot();
});

test('Snapshot test of Registration Container with errors', async () => {
  const tree = render(<RegistrationContainer />);
  const button = tree.getByText("Register")
  await userEvent.click(button);
  expect(tree).toMatchSnapshot();
});

test('Test that the password and verify password verification gives error on different passwords', async () => {
  const tree = render(<RegistrationContainer />);
  const passwordInput = tree.getByPlaceholderText('Password');
  const verifyPasswordInput = tree.getByPlaceholderText('Verify Password');
  await userEvent.type(passwordInput, "V4L!DPassword");
  await userEvent.type(verifyPasswordInput, "InV4L!DPassword")
  const button = tree.getByText("Register")
  await userEvent.click(button);
  const errorMessage = tree.getAllByText("Verify Password must be match password");
  expect(errorMessage).toBeTruthy();
});

test('Test that the password and verify password verification gives no error on same passwords', async () => {
  const tree = render(<RegistrationContainer />);
  const passwordInput = tree.getByPlaceholderText('Password');
  const verifyPasswordInput = tree.getByPlaceholderText('Verify Password');
  await userEvent.type(passwordInput, "V4L!DPassword");
  await userEvent.type(verifyPasswordInput, "V4L!DPassword")
  const button = tree.getByText("Register")
  await userEvent.click(button);
  const errorMessage = tree.queryByText("Verify Password must be match password")
  expect(errorMessage).toBeNull();
});

test('Test that the name does not allow more than 24 characters', async () => {
  const tree = render(<RegistrationContainer />);
  const firstNameInput = tree.getByPlaceholderText('First Name');
  await userEvent.type(firstNameInput, "SUPERLONGNAMETHATSHOULDCAUSEFAILURE");
  const button = tree.getByText("Register")
  await userEvent.click(button);
  const errorMessage = tree.queryByText("The name must not exceed 24 characters")
  expect(errorMessage).toBeTruthy();
});
  
  
