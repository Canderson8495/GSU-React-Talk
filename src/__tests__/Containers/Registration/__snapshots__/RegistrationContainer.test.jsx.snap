// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of Registration Container 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Registration-container"
      >
        <div
          class="Registration-title"
        >
          <h1>
             Registration 
          </h1>
        </div>
        <div
          class="Registration-fields"
        >
          <div
            class="Registration-field"
          >
            <label
              for="fname"
            >
               
              First Name
               
            </label>
            <input
              class="Registration-input"
              id="fname"
              name="fname"
              placeholder="First Name"
              type="text"
              value=""
            />
            <p
              class="Registration-error"
            >
               
            </p>
          </div>
          <div
            class="Registration-field"
          >
            <label
              for="lname"
            >
               
              Last Name
               
            </label>
            <input
              class="Registration-input"
              id="lname"
              name="lname"
              placeholder="Last Name"
              type="text"
              value=""
            />
            <p
              class="Registration-error"
            >
               
            </p>
          </div>
          <div
            class="Registration-field"
          >
            <label
              for="email"
            >
               
              Email
               
            </label>
            <input
              class="Registration-input"
              id="email"
              name="email"
              placeholder="Email"
              type="text"
              value=""
            />
            <p
              class="Registration-error"
            >
               
            </p>
          </div>
          <div
            class="Registration-field"
          >
            <label
              for="password"
            >
               
              Password
               
            </label>
            <input
              class="Registration-input"
              id="password"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
            <p
              class="Registration-error"
            >
               
            </p>
          </div>
          <div
            class="Registration-field"
          >
            <label
              for="vpassword"
            >
               
              Verify Password
               
            </label>
            <input
              class="Registration-input"
              id="vpassword"
              name="vpassword"
              placeholder="Verify Password"
              type="password"
              value=""
            />
            <p
              class="Registration-error"
            >
               
            </p>
          </div>
          <div
            class="Registration-field"
          >
            <label
              for="bday"
            >
               
              Birthday
               
            </label>
            <input
              class="Registration-input"
              id="bday"
              name="bday"
              placeholder="Birthday"
              type="text"
              value=""
            />
            <p
              class="Registration-error"
            >
               
            </p>
          </div>
        </div>
        <button
          class="Registration-button"
          type="button"
        >
          <p>
             Register 
          </p>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Registration-container"
    >
      <div
        class="Registration-title"
      >
        <h1>
           Registration 
        </h1>
      </div>
      <div
        class="Registration-fields"
      >
        <div
          class="Registration-field"
        >
          <label
            for="fname"
          >
             
            First Name
             
          </label>
          <input
            class="Registration-input"
            id="fname"
            name="fname"
            placeholder="First Name"
            type="text"
            value=""
          />
          <p
            class="Registration-error"
          >
             
          </p>
        </div>
        <div
          class="Registration-field"
        >
          <label
            for="lname"
          >
             
            Last Name
             
          </label>
          <input
            class="Registration-input"
            id="lname"
            name="lname"
            placeholder="Last Name"
            type="text"
            value=""
          />
          <p
            class="Registration-error"
          >
             
          </p>
        </div>
        <div
          class="Registration-field"
        >
          <label
            for="email"
          >
             
            Email
             
          </label>
          <input
            class="Registration-input"
            id="email"
            name="email"
            placeholder="Email"
            type="text"
            value=""
          />
          <p
            class="Registration-error"
          >
             
          </p>
        </div>
        <div
          class="Registration-field"
        >
          <label
            for="password"
          >
             
            Password
             
          </label>
          <input
            class="Registration-input"
            id="password"
            name="password"
            placeholder="Password"
            type="password"
            value=""
          />
          <p
            class="Registration-error"
          >
             
          </p>
        </div>
        <div
          class="Registration-field"
        >
          <label
            for="vpassword"
          >
             
            Verify Password
             
          </label>
          <input
            class="Registration-input"
            id="vpassword"
            name="vpassword"
            placeholder="Verify Password"
            type="password"
            value=""
          />
          <p
            class="Registration-error"
          >
             
          </p>
        </div>
        <div
          class="Registration-field"
        >
          <label
            for="bday"
          >
             
            Birthday
             
          </label>
          <input
            class="Registration-input"
            id="bday"
            name="bday"
            placeholder="Birthday"
            type="text"
            value=""
          />
          <p
            class="Registration-error"
          >
             
          </p>
        </div>
      </div>
      <button
        class="Registration-button"
        type="button"
      >
        <p>
           Register 
        </p>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshot test of Registration Container with errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Registration-container"
      >
        <div
          class="Registration-title"
        >
          <h1>
             Registration 
          </h1>
        </div>
        <div
          class="Registration-fields"
        >
          <div
            class="Registration-field"
          >
            <label
              for="fname"
            >
               
              First Name
               
            </label>
            <input
              class="Registration-input"
              id="fname"
              name="fname"
              placeholder="First Name"
              type="text"
              value=""
            />
            <p
              class="Registration-error"
            >
               
              This name must not be blank
            </p>
          </div>
          <div
            class="Registration-field"
          >
            <label
              for="lname"
            >
               
              Last Name
               
            </label>
            <input
              class="Registration-input"
              id="lname"
              name="lname"
              placeholder="Last Name"
              type="text"
              value=""
            />
            <p
              class="Registration-error"
            >
               
              This name must not be blank
            </p>
          </div>
          <div
            class="Registration-field"
          >
            <label
              for="email"
            >
               
              Email
               
            </label>
            <input
              class="Registration-input"
              id="email"
              name="email"
              placeholder="Email"
              type="text"
              value=""
            />
            <p
              class="Registration-error"
            >
               
              Email must not be blank
            </p>
          </div>
          <div
            class="Registration-field"
          >
            <label
              for="password"
            >
               
              Password
               
            </label>
            <input
              class="Registration-input"
              id="password"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
            <p
              class="Registration-error"
            >
               
              password must not be blank
            </p>
          </div>
          <div
            class="Registration-field"
          >
            <label
              for="vpassword"
            >
               
              Verify Password
               
            </label>
            <input
              class="Registration-input"
              id="vpassword"
              name="vpassword"
              placeholder="Verify Password"
              type="password"
              value=""
            />
            <p
              class="Registration-error"
            >
               
              Verify Password must not be blank
            </p>
          </div>
          <div
            class="Registration-field"
          >
            <label
              for="bday"
            >
               
              Birthday
               
            </label>
            <input
              class="Registration-input"
              id="bday"
              name="bday"
              placeholder="Birthday"
              type="text"
              value=""
            />
            <p
              class="Registration-error"
            >
               
              Birthday must not be blank
            </p>
          </div>
        </div>
        <button
          class="Registration-button"
          type="button"
        >
          <p>
             Register 
          </p>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Registration-container"
    >
      <div
        class="Registration-title"
      >
        <h1>
           Registration 
        </h1>
      </div>
      <div
        class="Registration-fields"
      >
        <div
          class="Registration-field"
        >
          <label
            for="fname"
          >
             
            First Name
             
          </label>
          <input
            class="Registration-input"
            id="fname"
            name="fname"
            placeholder="First Name"
            type="text"
            value=""
          />
          <p
            class="Registration-error"
          >
             
            This name must not be blank
          </p>
        </div>
        <div
          class="Registration-field"
        >
          <label
            for="lname"
          >
             
            Last Name
             
          </label>
          <input
            class="Registration-input"
            id="lname"
            name="lname"
            placeholder="Last Name"
            type="text"
            value=""
          />
          <p
            class="Registration-error"
          >
             
            This name must not be blank
          </p>
        </div>
        <div
          class="Registration-field"
        >
          <label
            for="email"
          >
             
            Email
             
          </label>
          <input
            class="Registration-input"
            id="email"
            name="email"
            placeholder="Email"
            type="text"
            value=""
          />
          <p
            class="Registration-error"
          >
             
            Email must not be blank
          </p>
        </div>
        <div
          class="Registration-field"
        >
          <label
            for="password"
          >
             
            Password
             
          </label>
          <input
            class="Registration-input"
            id="password"
            name="password"
            placeholder="Password"
            type="password"
            value=""
          />
          <p
            class="Registration-error"
          >
             
            password must not be blank
          </p>
        </div>
        <div
          class="Registration-field"
        >
          <label
            for="vpassword"
          >
             
            Verify Password
             
          </label>
          <input
            class="Registration-input"
            id="vpassword"
            name="vpassword"
            placeholder="Verify Password"
            type="password"
            value=""
          />
          <p
            class="Registration-error"
          >
             
            Verify Password must not be blank
          </p>
        </div>
        <div
          class="Registration-field"
        >
          <label
            for="bday"
          >
             
            Birthday
             
          </label>
          <input
            class="Registration-input"
            id="bday"
            name="bday"
            placeholder="Birthday"
            type="text"
            value=""
          />
          <p
            class="Registration-error"
          >
             
            Birthday must not be blank
          </p>
        </div>
      </div>
      <button
        class="Registration-button"
        type="button"
      >
        <p>
           Register 
        </p>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
